### DISCUSSION TITLE
Job start up times RunnerDeployment vs RunnerSet

### DISCUSSION THREAD

**Discussion by jespada** [actions/actions-runner-controller#3011] [Questions]
#### Context:
I've been experimenting with both `RunnerDeployment` and `RunnerSet` for job execution. While both are functional, I've observed a noticeable difference in job start-up times between the two.

#### Issue:
I've noticed that jobs initiated through `RunnerSet` tend to start up slower compared to those initiated via `RunnerDeployment` when using webhooks.

#### Questions:

1. Is it expected for `RunnerSet` to have slower job start-up times when compared to `RunnerDeployment`? If yes, could you provide some insights into why this might be the case? (webhook vs long polling?)
2. Does it make sense to use webhooks with `RunnerSet`? If so, are there any best practices or configurations to optimize start-up times?
3. Is there a way to improve the start-up times for jobs using `RunnerSet` to be more in line with `RunnerDeployment`?

#### Additional Information:
* I'm running the latest release of both RunnerDeployment and RunnerSet
* They both run with the same specs.
* RunnerSet is deployed using [runner-scale-set-controller chart](https://github.com/actions/actions-runner-controller/tree/master/charts/gha-runner-scale-set-controller) and [the runner-scale-set chart](https://github.com/actions/actions-runner-controller/tree/master/charts/gha-runner-scale-set)
* RunnerDeployment is deployed using [the actions controller chart](https://github.com/actions/actions-runner-controller/tree/master/charts/actions-runner-controller) and a the runners with the webhook  scaling method similar to what's described in [the docs](https://github.com/actions/actions-runner-controller/blob/master/docs/automatically-scaling-runners.md#webhook-driven-scaling)

Thank you for your time and assistance. Looking forward to some insights into this behavior!

**Comment by mahmud2011** [comment]
It's been a while you started the discussion and unfortunately no answer! Did you recently experiment this again?

**Reply by jespada** [reply]
Yes I can see it daily, I'm changing some of my config setup to see if the `RunnerSet` "issue" improves and will report back (it could be tied to the underlining VM startup times, but both of them run on the same node-pool..)
